org.eevolution.hr_and_payroll/src/main/java/base/org/eevolution/process/SendPayrollReceiptByEmail.java

```diff
@@ -0,0 +1,219 @@
+    }    //	sendEMail
+
+    /**
+     * Send Individual Mail
+     *
+     * @param mailText    Mail Test
+     * @param paymentId   user
+     * @param unSubscribe unsubscribe message
+     * @return true if mail has been sent
+     */
+    private Boolean sendIndividualMail(MMailText mailText, int paymentId, String unSubscribe) {
+        try {
+            MPayment payment = new MPayment(getCtx(), paymentId, get_TrxName());
+            if (payment == null)
+                throw new AdempiereException("@C_Payment_ID@ @NotFound@");
+
+            MBPartner employee = (MBPartner) payment.getC_BPartner();
+
+            String message = mailText.getMailText(true);
+            if (unSubscribe != null)
+                message += unSubscribe;
+
+            StringBuffer whereClause = new StringBuffer();
+            whereClause.append(MBPartnerLocation.COLUMNNAME_C_BPartner_ID)
+                    .append(" = ? AND ")
+                    .append(MBPartnerLocation.COLUMNNAME_ContactType)
+                    .append("=?");
+
+            MBPartnerLocation location = new Query(getCtx(), MBPartnerLocation.Table_Name, whereClause.toString(), get_TrxName())
+                    .setOnlyActiveRecords(true)
+                    .setParameters(employee.getC_BPartner_ID(), MBPartnerLocation.CONTACTTYPE_Primary)
+                    .first();
+
+            if (location == null) {
+                addLog(0, null, null, employee.getName() + " @Email@ @NotFound@");
+                return false;
+            }
+
+            MClient client = MClient.get(getCtx());
+            MUser user = MUser.get(getCtx(), Env.getAD_User_ID(getCtx()));
+            EMail email = client.createEMail(user, location.getEMail() , mailText.getMailHeader(), message);
+            if (mailText.isHtml())
+                email.setMessageHTML(mailText.getMailHeader(), message);
+            else {
+                email.setSubject(mailText.getMailHeader());
+                email.setMessageText(message);
+            }
+            File attachment = getPDF(paymentId);
+            if (attachment != null)
+                email.addAttachment(attachment);
+
+            if (!email.isValid() && !email.isValid(true)) {
+                addLog(0, null, null, employee.getName() + " @Email@ @NotValid@ " + email);
+                log.warning("NOT VALID - " + email);
+                return Boolean.FALSE;
+            }
+            boolean ok = EMail.SENT_OK.equals(email.send());
+            if (ok) {
+                addLog(0, null, null, employee.getName() + " @Email@ @OK@");
+                log.fine(employee.getURL());
+            } else
+                log.warning("FAILURE - " + employee.getURL());
+
+            addLog(0, null, null, (ok ? "@OK@" : "@ERROR@") + " - " + email.getFrom().getAddress());
+            return ok;
+        } catch (Exception e) {
+            return Boolean.FALSE;
+        }
+    }    //	sendIndividualMail
+
+
+    private int[] getPaymentIds() {
```

Comment:
It allway send all payments, is nice if can be filtered by EMail sent or a Date range